craft_grammar/_processor.py

```diff
@@ -53,12 +70,15 @@ def __init__(
                         returning true if it is valid.
         :param arch: the architecture the system is on.
         :param target_arch: the architecture the system is to build for.
+        :param platforms: the platforms to build. Duplicates are ignored.
```

Comment:
Since `for a,b` is invalid, what is the use-case for `platforms` having more than 1 item? Does this refer to all the possible platforms that a project declares, or just to the single platform that is currently being built?

Reply:
ah I see, I remember some of that conversation now. I think I tripped on "the platforms to build", it made me think of the full set of platforms. Maybe something like "the currently active platform identifiers" would be less confusing? Well, less confusing in this particular aspect I guess